File: test/new_lang/test12/TestProgram.pas
program TestProgram;
uses SplashKit;

procedure Main();
var
  s: Array of String;
  i: Integer;
begin
  for i := 0 to 10000 do
  begin
    s := GetStrings();
    PrintStrings(s);
    AddString(s, 'Yay');
    PrintStrings(s);
  end;
  ReadLn();
end;

begin
  Main();
end.

File: test/new_lang/test12/TestProgram.cs
using SplashKitSDK;
using static SplashKitSDK.SplashKit;
using static System.Console;
using System.Collections.Generic;

public class Program
{
  public static void Main()
  {
    WriteLine("Enter to start...");
    ReadLine();

    List<string> s = new List<string>();
    s.Add("Hello");

    for (int i = 0; i < 10000; i++)
    {
      // s = GetStrings();
      PrintStrings(s);
      // AddString(ref s, "Yay");
      // PrintStrings(s);
    }

    ReadLine();
  }
}
